client/chatroomwidget.cpp

```diff
@@ -658,6 +661,70 @@ void ChatRoomWidget::quote(const QString& htmlText)
     m_chatEdit->insertPlainText(sendString);
 }
 
+void ChatRoomWidget::showMenu(const QString& hoveredLink, bool showingDetails,
+                              int index)
+{
+    const QString eventType = m_messageModel->data(m_messageModel->index(index, 0),
```

Comment:
```suggestion
    const auto eventType = m_messageModel->index(index, 0)->data(
```
And you really should put `m_messageModel->index(index, 0)` to a separate variable, instead of calculating it 3 times.